common/src/Microsoft.Azure.IIoT.Hub.Module.Client/src/Default/MqttClient/MqttClientAdapter.cs

```diff
@@ -0,0 +1,617 @@
+                _product = product ?? throw new ArgumentNullException(nameof(product));
+                _deviceId = deviceId ?? throw new ArgumentNullException(nameof(deviceId));
+                _timeout = timeout;
+                _retry = retry;
+                _onConnectionLost = onConnectionLost ?? throw new ArgumentNullException(nameof(onConnectionLost));
+                _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+                _messageExpiryInterval = messageExpiryInterval;
+                _qualityOfServiceLevel = qualityOfServiceLevel;
+                _useIoTHubTopics = useIoTHubTopics;
+                _telemetryTopicTemplate = telemetryTopicTemplate;
+            }
+
+            /// <summary>
+            /// Create and connect an instance of the client adapter.
+            /// </summary>
+            /// <param name="client">MQTT server client.</param>
+            /// <param name="product">Custom product information.</param>
+            /// <param name="cs">Connection string for the MQTT server.</param>
+            /// <param name="deviceId">Id of the device.</param>
+            /// <param name="telemetryTopicTemplate">Telemetry topic template.</param>
+            /// <param name="timeout">Timeout used for operations.</param>
+            /// <param name="retry">Retry policy used for operations.</param>
+            /// <param name="onConnectionLost">Handler for when the MQTT server connection is lost.</param>
+            /// <param name="logger">Logger used for operations</param>
+            /// <returns></returns>
+            public static async Task<IClient> CreateAsync(
+                IManagedMqttClient client,
+                string product,
+                MqttClientConnectionStringBuilder cs,
+                string deviceId,
+                string telemetryTopicTemplate,
+                TimeSpan timeout,
+                IRetryPolicy retry,
+                Action onConnectionLost,
+                ILogger logger) {
+                var options = new MqttClientOptionsBuilder()
+                    .WithClientId(cs.DeviceId)
+                    .WithTcpServer(tcpOptions => {
+                        tcpOptions.Server = cs.HostName;
+                        tcpOptions.Port = cs.Port;
+
+                        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) {
+                            tcpOptions.BufferSize = 64 * 1024;
+                        }
+                    });
+
+                if (!string.IsNullOrWhiteSpace(cs.Username) && !string.IsNullOrWhiteSpace(cs.Password)) {
+                    options = options.WithCredentials(cs.Username, cs.Password);
+                }
+
+                var tls_options = new MqttClientOptionsBuilderTlsParameters {
+                    UseTls = cs.UsingIoTHub,
+                    SslProtocol = SslProtocols.Tls12,
+                    IgnoreCertificateRevocationErrors = true
```

Comment:
is this safe to hardcode?

Reply:
Let's take this down as bug and discuss.  I would say this should never be true unless you override it through an option in the connection string.  @jschweda, why was this chosen to be true?